pkg/pidfile/pidfile.go

```diff
@@ -0,0 +1,75 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+package pidfile
+
+import (
+	"bytes"
+	"errors"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/cilium/tetragon/pkg/defaults"
+	"github.com/cilium/tetragon/pkg/option"
+	"github.com/cilium/tetragon/pkg/reader/proc"
+)
+
+var (
+	ErrPidFileNotPresent = errors.New("pid file is not present")
+	ErrPidIsNotAlive     = errors.New("process is not alive")
+	ErrPidIsStillAlive   = errors.New("process is already running")
+)
+
+func readPidFile() (uint64, error) {
+	_, err := os.Stat(defaults.DefaultPidFile)
+	if err != nil {
+		return 0, ErrPidFileNotPresent
+	}
+
+	data, err := os.ReadFile(defaults.DefaultPidFile)
+	if err != nil {
+		return 0, err
+	}
+
+	pid := string(bytes.TrimSpace(data))
+	_, err = os.Stat(filepath.Join(option.Config.ProcFS, pid))
+	if err != nil {
+		return 0, ErrPidIsNotAlive
+	}
+
+	return strconv.ParseUint(pid, 10, 32)
+}
+
+// CreatePidFile() Creates a Pid file
+//
+// On success returns:
+//
+//	The current PID and nil on success
+//
+// On failure returns:
+//
+//	Returns old pid if still running and PidIsStillAlive error
+//	Or returns zero and an error
+func CreatePidFile() (uint64, error) {
+	pid, err := readPidFile()
+	if err == nil && pid != 0 {
+		return pid, ErrPidIsStillAlive
+	}
+
+	if err != nil &&
+		errors.Is(err, ErrPidFileNotPresent) == false &&
+		errors.Is(err, ErrPidIsNotAlive) == false {
+		return 0, err
+	}
+
+	pid, err = proc.GetSelfPid(option.Config.ProcFS)
```

Comment:
I think you could use standard lib https://pkg.go.dev/os#Getpid here as well to simplify and remove the function from `proc` package.

Reply:
Since we export the procfs so I want to go with it, and avoid any pid namespace foo that could mess up the getpid since usually it works with namespaces ;-)

Reply:
ah I see! completely out of subject but speaking about this stuff, I read `exe` in /proc in the matchBinaries rework on the userside as root, is there anything about namespace that I should be concerned about?